### PR TITLE
InCanvasSearch: Place node where user entered

### PR DESCRIPTION
**Pull request by aosyatnik**
### Purpose

Link to YouTrack:
[MAGN-7410](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7410) Place node at point in workspace where user entered in-canvas search

Node is created at the same place, where InCanvasSearch was located.
image

image

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [ ] The level of testing this PR includes is appropriate

### Reviewers

@pboyer 
@Benglin 

### FYIs

@lillismith
@kronz

### CODE REVIEW

File: src/DynamoCoreWpf/Controls/IncanvasSearchControl.xaml.cs [hunk 1]
```diff
@@ -33,9 +36,17 @@ private SearchViewModel ViewModel
             get { return DataContext as SearchViewModel; }
         }
 
+        private WorkspaceView WorkspaceView;
```
Comment (Benglin) [inline src/DynamoCoreWpf/Controls/IncanvasSearchControl.xaml.cs]: Hi @aosyatnik, minor thing but private fields should have names such as `workspaceView`.

File: src/DynamoCoreWpf/Controls/IncanvasSearchControl.xaml.cs [hunk 2]
```diff
@@ -70,6 +81,9 @@ private void ExecuteSearchElement(ListBoxItem listBoxItem)
             var searchElement = listBoxItem.DataContext as NodeSearchElementViewModel;
             if (searchElement != null)
             {
+                Point targetLocation = this.TranslatePoint(new Point(0, 0), WorkspaceView);
+
+                searchElement.Position = targetLocation;
```
Comment (Benglin) [inline src/DynamoCoreWpf/Controls/IncanvasSearchControl.xaml.cs]: Could have been this, but just cosmetic :)

`searchElement.Position = TranslatePoint(new Point(), workspaceView);`

File: src/DynamoCoreWpf/ViewModels/Search/NodeSearchElementViewModel.cs [hunk 3]
```diff
@@ -169,15 +168,17 @@ public ImageSource LargeIcon
             }
         }
 
+        public Point Position;
+
```
Comment (Benglin) [inline src/DynamoCoreWpf/ViewModels/Search/NodeSearchElementViewModel.cs]: This should have been made a property instead, just like `ClickedCommand` below it, and then made `internal`.

**Comment by Benglin** [comment]
Thanks @aosyatnik, minor cosmetic changes and we'll be good to go :)

**Comment by aosyatnik** [comment]
Hi, @Benglin , thank you for review. I've updated places, those you mentioned.

**Comment by kronz** [comment]
@Benglin @pboyer is this good to go?  Looks like it is approved?

**Comment by Benglin** [comment]
Thanks for fixing them, @aosyatnik! LGTM

**Pull request merged**